diff --git a/configure b/configure
index 17596942749abe569c7d42f8abc1b48ce88cb148..c628aa8b92109f9cedaa8e5a70c62282f9ea79ff 100755
--- a/configure
+++ b/configure
@@ -5793,7 +5793,6 @@ elif enabled mips; then
         disable mipsdsp
         disable mipsdspr2
         # Disable DSP stuff for generic CPU, it can't be detected at runtime.
-        warn 'generic cpu selected'
     fi
 
 elif enabled ppc; then
@@ -7182,7 +7181,6 @@ if ! disabled pthreads && ! enabled w32threads && ! enabled os2threads; then
     fi
 fi
 
-enabled  zlib && { check_pkg_config zlib zlib "zlib.h" zlibVersion ||
-                   check_lib zlib   zlib.h      zlibVersion    -lz; }
+enabled  zlib && enable zlib
 enabled bzlib && check_lib bzlib bzlib.h BZ2_bzlibVersion    -lbz2
 enabled  lzma && check_lib lzma   lzma.h lzma_version_number -llzma
